src/mono/browser/runtime/loader/config.ts

```diff
@@ -283,24 +286,41 @@ export function isDebuggingSupported (): boolean {
 async function loadBootConfig (module: DotnetModuleInternal): Promise<void> {
     const defaultConfigSrc = loaderHelpers.locateFile(module.configSrc!);
 
-    const loaderResponse = loaderHelpers.loadBootResource !== undefined ?
-        loaderHelpers.loadBootResource("manifest", "blazor.boot.json", defaultConfigSrc, "", "manifest") :
-        defaultLoadBootConfig(defaultConfigSrc);
-
-    let loadConfigResponse: Response;
+    let loaderResponse = null;
+    if (loaderHelpers.loadBootResource !== undefined) {
+        loaderResponse = loaderHelpers.loadBootResource("manifest", "blazor.boot.json", defaultConfigSrc, "", "manifest");
+    }
 
+    let loadedConfigResponse: Response | null = null;
+    let loadedConfig: MonoConfig;
     if (!loaderResponse) {
-        loadConfigResponse = await defaultLoadBootConfig(appendUniqueQuery(defaultConfigSrc, "manifest"));
+        if (defaultConfigSrc.includes(".json")) {
+            loadedConfigResponse = await fetchBootConfig(appendUniqueQuery(defaultConfigSrc, "manifest"));
+            loadedConfig = await readBootConfigResponse(loadedConfigResponse);
+        } else {
+            loadedConfig = (await import(appendUniqueQuery(defaultConfigSrc, "manifest"))).config;
```

Comment:
Is this because we are letting people control where `blazor.boot.json` was being loaded from?

Reply:
@maraf is there a need for us to support that moving forward?

What scenario do we think still mandates we keep this around.

I'm saying this because the more we can simplify this process the better. 

Caching and all those things we are handling through the importmap work we did, and I can't really see a reason for this being loaded from elsewhere

Reply:
I think this could be in use by non-blazor customers, but I don't have anything specific

Reply:
What was the original motivation? Modifying the URLs to point to some CDN or subpath on the server is still valid. 
This callback process all framework assets, not just JS modules

Reply:
Yes, the original motivation was for letting people control the URLs of the deployed app, when we had issues with the anti-virus.

I think it's still fine for things that are "data" like the webcil, but is there a reason why we couldn't rely on relative paths for the module bits? After all, you can already change these urls during the build process modifying the relative paths of the respective assets.

Reply:
I don't have a strong opinion. Since blazor users can configure .NET runtime through a callback on `Blazor.start`, they have all the flexibility to define the assets and their URLs in a defferent way than through the loadResource callback, on the other it's a breaking change.